src/serializer/factorify.js

```diff
@@ -92,19 +92,44 @@ function getObjectKeys(obj: BabelNodeObjectExpression): string | false {
 export function factorifyObjects(body: Array<BabelNodeStatement>, factoryNameGenerator: NameGenerator) {
   let signatures = Object.create(null);
 
+  let initializerAstNodeName = "init";
   for (let node of body) {
-    if (node.type !== "VariableDeclaration") continue;
+    switch (node.type) {
+      case "VariableDeclaration":
+        for (let declar of node.declarations) {
+          let { init } = declar;
+          if (!init) continue;
+          if (init.type !== "ObjectExpression") continue;
+
+          let keys = getObjectKeys(init);
+          if (!keys) continue;
+
+          let declars = (signatures[keys] = signatures[keys] || []);
+          declars.push(declar);
+        }
+        break;
+
+      case "ExpressionStatement":
+        const expr = node.expression;
+        if (expr.type !== "AssignmentExpression") {
+          break;
+        }
+        const { right } = expr;
+        if (right.type !== "ObjectExpression") {
+          break;
+        }
 
-    for (let declar of node.declarations) {
-      let { init } = declar;
-      if (!init) continue;
-      if (init.type !== "ObjectExpression") continue;
+        let keys = getObjectKeys(right);
+        if (!keys) continue;
 
-      let keys = getObjectKeys(init);
-      if (!keys) continue;
+        initializerAstNodeName = "right";
+        let declars = (signatures[keys] = signatures[keys] || []);
+        declars.push(node.expression);
+        break;
 
-      let declars = (signatures[keys] = signatures[keys] || []);
-      declars.push(declar);
+      default:
```

Comment:
Seems useless. If you added the `default` case to indicate that it wasn't just forgotten, then this should come with a lint rule that enforces `default` cases.